packages/core/schematics/ng-generate/standalone-migration/standalone-bootstrap.ts

```diff
@@ -0,0 +1,630 @@
+    nodeLookup = nodeLookup || getNodeLookup(moduleSourceFile);
+    migrateImportsInBootstrapCall(
+        sourceFile, imports, nodeLookup, moduleImportsInNewCall, providersInNewCall, tracker,
+        nodesToCopy, languageService, typeChecker);
+  }
+
+  if (nodesToCopy.size > 0) {
+    let text = '\n\n';
+    nodesToCopy.forEach(node => {
+      const transformedNode = remapDynamicImports(sourceFile.fileName, node);
+
+      // Use `getText` to try an preserve the original formatting. This only works if the node
+      // hasn't been transformed. If it has, we have to fall back to the printer.
+      if (transformedNode === node) {
+        text += transformedNode.getText() + '\n';
+      } else {
+        text += printer.printNode(ts.EmitHint.Unspecified, transformedNode, node.getSourceFile());
+      }
+    });
+    text += '\n';
+    tracker.insertText(sourceFile, getLastImportEnd(sourceFile), text);
+  }
+
+  replaceBootstrapCall(analysis, providersInNewCall, moduleImportsInNewCall, tracker);
+}
+
+/**
+ * Replaces a `bootstrapModule` call with `bootstrapApplication`.
+ * @param analysis Analysis result of the `bootstrapModule` call.
+ * @param providers Providers that should be added to the new call.
+ * @param modules Modules that are being imported into the new call.
+ * @param tracker Object keeping track of the changes to the different files.
+ */
+function replaceBootstrapCall(
+    analysis: BootstrapCallAnalysis, providers: ts.Expression[], modules: ts.Expression[],
+    tracker: ChangeTracker): void {
+  const sourceFile = analysis.call.getSourceFile();
+  const componentPath =
+      getRelativeImportPath(sourceFile.fileName, analysis.component.getSourceFile().fileName);
+  const args = [tracker.addImport(sourceFile, analysis.component.name.text, componentPath)];
+  const bootstrapExpression =
+      tracker.addImport(sourceFile, 'bootstrapApplication', '@angular/platform-browser');
+
+  if (providers.length > 0 || modules.length > 0) {
+    const combinedProviders: ts.Expression[] = [];
+
+    if (modules.length > 0) {
+      const importProvidersExpression =
+          tracker.addImport(sourceFile, 'importProvidersFrom', '@angular/core');
+      combinedProviders.push(
+          ts.factory.createCallExpression(importProvidersExpression, [], modules));
+    }
+
+    // Push the providers after `importProvidersFrom` call for better readability.
+    combinedProviders.push(...providers);
+    const initializer = remapDynamicImports(
+        sourceFile.fileName,
+        ts.factory.createArrayLiteralExpression(combinedProviders, combinedProviders.length > 1));
+
+    args.push(ts.factory.createObjectLiteralExpression(
+        [ts.factory.createPropertyAssignment('providers', initializer)], true));
+  }
+
+  tracker.replaceNode(
+      analysis.call, ts.factory.createCallExpression(bootstrapExpression, [], args),
+      // Note: it's important to pass in the metadata source file! Otherwise TS won't print out
```

Comment:
Optional nit which I believe is easier to understand:

```suggestion
      // Note: it's important to pass in the source file where the provider expressions are taken from. [...]
```